File: Problem_Solving/Algorithms/Implementation/1_Grading_Students/Solution.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {

    public static final int MINIMUM_GRADE=38;

    public static List<Integer> gradingStudents(List<Integer> grades) {
        int gradesSize=grades.size();
        List<Integer> roundedGrades=new ArrayList<Integer>(gradesSize);

        for(int i=0;i<gradesSize;i++){
            //round if distance<=2 from ith grade to next multiple of 5
            if(grades.get(i)>=MINIMUM_GRADE && grades.get(i)%5>=3)
                roundedGrades.add(grades.get(i)-(grades.get(i)%5)+5);
            else
                roundedGrades.add(grades.get(i));
        }

        return roundedGrades;
    }

}

public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        int gradesCount = Integer.parseInt(bufferedReader.readLine().trim());

        List<Integer> grades = IntStream.range(0, gradesCount).mapToObj(i -> {
            try {
                return bufferedReader.readLine().replaceAll("\\s+$", "");
            } catch (IOException ex) {
                throw new RuntimeException(ex);
            }
        })
            .map(String::trim)
            .map(Integer::parseInt)
            .collect(toList());

        List<Integer> result = Result.gradingStudents(grades);

        bufferedWriter.write(
            result.stream()
                .map(Object::toString)
                .collect(joining("\n"))
        );

        bufferedReader.close();
        bufferedWriter.close();
    }
}

File: Problem_Solving/Algorithms/Implementation/1_Grading_Students/Solution.py
import math
import os
import random
import re
import sys

def gradingStudents(grades):
    minimumGrade=38
    roundedGrades=[]

    for k,v in enumerate(grades):
        #round if distance<=2 from ith grade to next multiple of 5
        if v>=minimumGrade and v%5>=3:
            roundedGrades.append(v-(v%5)+5)
        else:
            roundedGrades.append(v)

    return roundedGrades

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    grades_count = int(input().strip())

    grades = []

    for _ in range(grades_count):
        grades_item = int(input().strip())
        grades.append(grades_item)

    result = gradingStudents(grades)

    fptr.write('\n'.join(map(str, result)))
    fptr.write('\n')

    fptr.close()
